### PR TITLE
feat: add FoldingHeading extension

### PR DESCRIPTION
**Pull request by d3m1d0v**

### CODE REVIEW

**Comment by d3m1d0v** [comment]
TODO: add docs how to connect FoldingHeading extension

File: src/extensions/yfm/YfmHeading/YfmHeadingSpecs/const.ts [hunk 1]
```diff
@@ -7,5 +7,5 @@ export const YfmHeadingAttr = {
     Level: headingLevelAttr,
     Id: 'id',
     DataLine: 'data-line',
-    Folding: 'folding',
+    Folding: 'folded',
```
[the remark this answers is not in the document]
Comment (d3m1d0v) [reply src/extensions/yfm/YfmHeading/YfmHeadingSpecs/const.ts:10]: "folded" explains the meaning better. That can be true if heading is folded (section is collapsed). False, if heading is open. And null if it is a common heading.

Field "Folding" remains the same for backward compatibility.

File: src/extensions/yfm/FoldingHeading/plugins/Folding.ts [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+                  class: 'pm-h-folding-content',
+                  nodeName: 'div',
+              };
+        decorations.push(Decoration.node(item.from, item.to, attrs));
+    }
+    for (const item of Object.values(separatorDecos)) {
+        decorations.push(
+            Decoration.node(item.from, item.to, {
+                nodeName: 'div',
+                class: 'pm-h-folding-label',
```
[the remark this answers is not in the document]
Comment (d3m1d0v) [reply src/extensions/yfm/FoldingHeading/plugins/Folding.ts:196]: > Also do we need to put pm in front of the class name in our code? I thought pm is the prefix that prosemirror puts in its own code.

`prosemirror` uses prefix `ProseMirror-` for its own classes.
I don't see anything wrong with using the prefix `pm-` to denote entities that can only be inside the wysiwyg editor.

**Comment by makhnatkin** [comment]
Screenshot 2024-07-31 at 14 01 48

The separator was inserted in the wrong place

**Review by makhnatkin — APPROVED** [review]

**Pull request merged**